docs/development/design/refactor-remote-repository.rst

```diff
@@ -0,0 +1,101 @@
+======================================
+ Refactor ``RemoteRepository`` object
+======================================
+
+
+This document describes the current usage of ``RemoteRepository`` objects and proposes a new normalized modeling.
+
+
+Goals
+=====
+
+* De-duplicate data stored in our database.
+* Save only one ``RemoteRepository`` per GitHub repository.
+* Use an intermediate table between ``RemoteRepository`` and ``User`` to store associated remote data for the specific user.
+* Make this model usable from our SSO implementation.
```

Comment:
This implies adding a new field `remote_id=` to make queries faster and do not depend on `full_name`.